vcd/resource_vcd_lb_app_rule.go

```diff
@@ -0,0 +1,220 @@
+		},
+
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "vCD organization in which the Application Rule is located",
+			},
+			"vdc": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "vCD virtual datacenter in which the Application Rule is located",
+			},
+			"edge_gateway": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Edge gateway name in which the Application Rule is located",
+			},
+			"name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Unique Application Rule name",
+			},
+			"script": &schema.Schema{
+				Required: true,
+				Type:     schema.TypeList,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Description: "The script for the application rule. Each line as a separate array element",
+			},
+		},
+	}
+}
+
+func resourceVcdLBAppRuleCreate(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+	vcdClient.lockParentEdgeGtw(d)
+	defer vcdClient.unLockParentEdgeGtw(d)
+
+	edgeGateway, err := vcdClient.GetEdgeGatewayFromResource(d, "edge_gateway")
+	if err != nil {
+		return fmt.Errorf(errorUnableToFindEdgeGateway, err)
+	}
+
+	LBRule, err := getLBAppRuleType(d)
+	if err != nil {
+		return fmt.Errorf("unable to expand load balancer app rule: %s", err)
+	}
+
+	createdPool, err := edgeGateway.CreateLBAppRule(LBRule)
+	if err != nil {
+		return fmt.Errorf("error creating new load balancer app rule: %s", err)
+	}
+
+	// We store the values once again because response include pool member IDs
+	err = setLBAppRuleData(d, createdPool)
+	if err != nil {
+		return err
+	}
+	d.SetId(createdPool.ID)
+	return nil
+}
+
+func resourceVcdLBAppRuleRead(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+
+	edgeGateway, err := vcdClient.GetEdgeGatewayFromResource(d, "edge_gateway")
+	if err != nil {
+		return fmt.Errorf(errorUnableToFindEdgeGateway, err)
+	}
+
+	readLBRule, err := edgeGateway.ReadLBAppRuleByID(d.Id())
+	if err != nil {
+		d.SetId("")
+		return fmt.Errorf("unable to find load balancer app rule with ID %s: %s", d.Id(), err)
+	}
+
+	return setLBAppRuleData(d, readLBRule)
+}
+
+func resourceVcdLBAppRuleUpdate(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+	vcdClient.lockParentEdgeGtw(d)
+	defer vcdClient.unLockParentEdgeGtw(d)
+
+	edgeGateway, err := vcdClient.GetEdgeGatewayFromResource(d, "edge_gateway")
+	if err != nil {
+		return fmt.Errorf(errorUnableToFindEdgeGateway, err)
+	}
+
+	updateLBRuleConfig, err := getLBAppRuleType(d)
+	if err != nil {
+		return fmt.Errorf("could not expand load balancer app rule for update: %s", err)
+	}
+
+	updatedLBRule, err := edgeGateway.UpdateLBAppRule(updateLBRuleConfig)
```

Comment:
Again, Struct for name and script

Reply:
Put a comment above.

Reply:
don't see the change

Reply:
Sorry. I meant a comment in the above discussion we have. https://github.com/terraform-providers/terraform-provider-vcd/pull/278#discussion_r301886652

Reply:
Oh, Go ti now:)